2025-07-10 13:07:13,111 - worker - INFO - --- Starting BlogSpy Worker ---
2025-07-10 13:07:13,426 - worker - INFO - Model loaded successfully.
2025-07-10 13:07:13,426 - worker - INFO - Initialized with BATCH_SIZE=500 and max_workers=80.
2025-07-10 13:07:21,631 - worker - ERROR - Reaper: Failed to reset stalled jobs: (psycopg2.errors.UndefinedColumn) column urls.locked_at does not exist
LINE 1: ...RE urls.status IN ('classifying', 'crawling') AND urls.locke...
                                                             ^

[SQL: UPDATE urls SET status=CASE WHEN (urls.status = %(status_1)s) THEN %(param_1)s WHEN (urls.status = %(status_2)s) THEN %(param_2)s END, locked_at=%(locked_at)s WHERE urls.status IN (%(status_3_1)s, %(status_3_2)s) AND urls.locked_at < %(locked_at_1)s]
[parameters: {'status_1': 'classifying', 'param_1': 'pending_classification', 'status_2': 'crawling', 'param_2': 'pending_crawl', 'locked_at': None, 'locked_at_1': datetime.datetime(2025, 7, 10, 7, 22, 13, 530626, tzinfo=datetime.timezone.utc), 'status_3_1': 'classifying', 'status_3_2': 'crawling'}]
(Background on this error at: https://sqlalche.me/e/20/f405)
Traceback (most recent call last):
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1963, in _exec_single_context
    self.dialect.do_execute(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/default.py", line 943, in do_execute
    cursor.execute(statement, parameters)
psycopg2.errors.UndefinedColumn: column urls.locked_at does not exist
LINE 1: ...RE urls.status IN ('classifying', 'crawling') AND urls.locke...
                                                             ^


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/ayush/Desktop/code/blogspy_predictor/src/worker.py", line 289, in reset_stalled_jobs
    rows_reset = stalled_jobs_query.update(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/query.py", line 3299, in update
    result: CursorResult[Any] = self.session.execute(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 2365, in execute
    return self._execute_internal(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 2251, in _execute_internal
    result: Result[Any] = compile_state_cls.orm_execute_statement(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/bulk_persistence.py", line 1648, in orm_execute_statement
    return super().orm_execute_statement(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/context.py", line 306, in orm_execute_statement
    result = conn.execute(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1415, in execute
    return meth(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/sql/elements.py", line 523, in _execute_on_connection
    return connection._execute_clauseelement(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1637, in _execute_clauseelement
    ret = self._execute_context(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1842, in _execute_context
    return self._exec_single_context(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1982, in _exec_single_context
    self._handle_dbapi_exception(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 2351, in _handle_dbapi_exception
    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1963, in _exec_single_context
    self.dialect.do_execute(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/default.py", line 943, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedColumn) column urls.locked_at does not exist
LINE 1: ...RE urls.status IN ('classifying', 'crawling') AND urls.locke...
                                                             ^

[SQL: UPDATE urls SET status=CASE WHEN (urls.status = %(status_1)s) THEN %(param_1)s WHEN (urls.status = %(status_2)s) THEN %(param_2)s END, locked_at=%(locked_at)s WHERE urls.status IN (%(status_3_1)s, %(status_3_2)s) AND urls.locked_at < %(locked_at_1)s]
[parameters: {'status_1': 'classifying', 'param_1': 'pending_classification', 'status_2': 'crawling', 'param_2': 'pending_crawl', 'locked_at': None, 'locked_at_1': datetime.datetime(2025, 7, 10, 7, 22, 13, 530626, tzinfo=datetime.timezone.utc), 'status_3_1': 'classifying', 'status_3_2': 'crawling'}]
(Background on this error at: https://sqlalche.me/e/20/f405)
2025-07-10 13:07:23,679 - worker - ERROR - An unexpected error occurred in the main loop: (psycopg2.errors.UndefinedColumn) column "locked_at" of relation "urls" does not exist
LINE 1: UPDATE urls SET status='crawling', locked_at='2025-07-10T07:...
                                           ^

[SQL: UPDATE urls SET status=%(status)s, locked_at=%(locked_at)s WHERE urls.id IN (%(id_1_1)s, %(id_1_2)s)]
[parameters: {'status': 'crawling', 'locked_at': datetime.datetime(2025, 7, 10, 7, 37, 23, 169516, tzinfo=datetime.timezone.utc), 'id_1_1': 1, 'id_1_2': 2}]
(Background on this error at: https://sqlalche.me/e/20/f405)
Traceback (most recent call last):
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1963, in _exec_single_context
    self.dialect.do_execute(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/default.py", line 943, in do_execute
    cursor.execute(statement, parameters)
psycopg2.errors.UndefinedColumn: column "locked_at" of relation "urls" does not exist
LINE 1: UPDATE urls SET status='crawling', locked_at='2025-07-10T07:...
                                           ^


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/ayush/Desktop/code/blogspy_predictor/src/worker.py", line 347, in main
    db.query(URL).filter(URL.id.in_(job_ids)).update(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/query.py", line 3299, in update
    result: CursorResult[Any] = self.session.execute(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 2365, in execute
    return self._execute_internal(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 2251, in _execute_internal
    result: Result[Any] = compile_state_cls.orm_execute_statement(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/bulk_persistence.py", line 1648, in orm_execute_statement
    return super().orm_execute_statement(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/context.py", line 306, in orm_execute_statement
    result = conn.execute(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1415, in execute
    return meth(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/sql/elements.py", line 523, in _execute_on_connection
    return connection._execute_clauseelement(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1637, in _execute_clauseelement
    ret = self._execute_context(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1842, in _execute_context
    return self._exec_single_context(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1982, in _exec_single_context
    self._handle_dbapi_exception(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 2351, in _handle_dbapi_exception
    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1963, in _exec_single_context
    self.dialect.do_execute(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/default.py", line 943, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedColumn) column "locked_at" of relation "urls" does not exist
LINE 1: UPDATE urls SET status='crawling', locked_at='2025-07-10T07:...
                                           ^

[SQL: UPDATE urls SET status=%(status)s, locked_at=%(locked_at)s WHERE urls.id IN (%(id_1_1)s, %(id_1_2)s)]
[parameters: {'status': 'crawling', 'locked_at': datetime.datetime(2025, 7, 10, 7, 37, 23, 169516, tzinfo=datetime.timezone.utc), 'id_1_1': 1, 'id_1_2': 2}]
(Background on this error at: https://sqlalche.me/e/20/f405)
2025-07-10 13:07:30,437 - worker - ERROR - Reaper: Failed to reset stalled jobs: (psycopg2.errors.UndefinedColumn) column urls.locked_at does not exist
LINE 1: ...RE urls.status IN ('classifying', 'crawling') AND urls.locke...
                                                             ^

[SQL: UPDATE urls SET status=CASE WHEN (urls.status = %(status_1)s) THEN %(param_1)s WHEN (urls.status = %(status_2)s) THEN %(param_2)s END, locked_at=%(locked_at)s WHERE urls.status IN (%(status_3_1)s, %(status_3_2)s) AND urls.locked_at < %(locked_at_1)s]
[parameters: {'status_1': 'classifying', 'param_1': 'pending_classification', 'status_2': 'crawling', 'param_2': 'pending_crawl', 'locked_at': None, 'locked_at_1': datetime.datetime(2025, 7, 10, 7, 22, 29, 197471, tzinfo=datetime.timezone.utc), 'status_3_1': 'classifying', 'status_3_2': 'crawling'}]
(Background on this error at: https://sqlalche.me/e/20/f405)
Traceback (most recent call last):
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1963, in _exec_single_context
    self.dialect.do_execute(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/default.py", line 943, in do_execute
    cursor.execute(statement, parameters)
psycopg2.errors.UndefinedColumn: column urls.locked_at does not exist
LINE 1: ...RE urls.status IN ('classifying', 'crawling') AND urls.locke...
                                                             ^


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/ayush/Desktop/code/blogspy_predictor/src/worker.py", line 289, in reset_stalled_jobs
    rows_reset = stalled_jobs_query.update(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/query.py", line 3299, in update
    result: CursorResult[Any] = self.session.execute(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 2365, in execute
    return self._execute_internal(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 2251, in _execute_internal
    result: Result[Any] = compile_state_cls.orm_execute_statement(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/bulk_persistence.py", line 1648, in orm_execute_statement
    return super().orm_execute_statement(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/context.py", line 306, in orm_execute_statement
    result = conn.execute(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1415, in execute
    return meth(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/sql/elements.py", line 523, in _execute_on_connection
    return connection._execute_clauseelement(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1637, in _execute_clauseelement
    ret = self._execute_context(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1842, in _execute_context
    return self._exec_single_context(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1982, in _exec_single_context
    self._handle_dbapi_exception(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 2351, in _handle_dbapi_exception
    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1963, in _exec_single_context
    self.dialect.do_execute(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/default.py", line 943, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedColumn) column urls.locked_at does not exist
LINE 1: ...RE urls.status IN ('classifying', 'crawling') AND urls.locke...
                                                             ^

[SQL: UPDATE urls SET status=CASE WHEN (urls.status = %(status_1)s) THEN %(param_1)s WHEN (urls.status = %(status_2)s) THEN %(param_2)s END, locked_at=%(locked_at)s WHERE urls.status IN (%(status_3_1)s, %(status_3_2)s) AND urls.locked_at < %(locked_at_1)s]
[parameters: {'status_1': 'classifying', 'param_1': 'pending_classification', 'status_2': 'crawling', 'param_2': 'pending_crawl', 'locked_at': None, 'locked_at_1': datetime.datetime(2025, 7, 10, 7, 22, 29, 197471, tzinfo=datetime.timezone.utc), 'status_3_1': 'classifying', 'status_3_2': 'crawling'}]
(Background on this error at: https://sqlalche.me/e/20/f405)
2025-07-10 13:07:32,485 - worker - ERROR - An unexpected error occurred in the main loop: (psycopg2.errors.UndefinedColumn) column "locked_at" of relation "urls" does not exist
LINE 1: UPDATE urls SET status='crawling', locked_at='2025-07-10T07:...
                                           ^

[SQL: UPDATE urls SET status=%(status)s, locked_at=%(locked_at)s WHERE urls.id IN (%(id_1_1)s, %(id_1_2)s)]
[parameters: {'status': 'crawling', 'locked_at': datetime.datetime(2025, 7, 10, 7, 37, 31, 871244, tzinfo=datetime.timezone.utc), 'id_1_1': 1, 'id_1_2': 2}]
(Background on this error at: https://sqlalche.me/e/20/f405)
Traceback (most recent call last):
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1963, in _exec_single_context
    self.dialect.do_execute(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/default.py", line 943, in do_execute
    cursor.execute(statement, parameters)
psycopg2.errors.UndefinedColumn: column "locked_at" of relation "urls" does not exist
LINE 1: UPDATE urls SET status='crawling', locked_at='2025-07-10T07:...
                                           ^


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/ayush/Desktop/code/blogspy_predictor/src/worker.py", line 347, in main
    db.query(URL).filter(URL.id.in_(job_ids)).update(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/query.py", line 3299, in update
    result: CursorResult[Any] = self.session.execute(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 2365, in execute
    return self._execute_internal(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 2251, in _execute_internal
    result: Result[Any] = compile_state_cls.orm_execute_statement(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/bulk_persistence.py", line 1648, in orm_execute_statement
    return super().orm_execute_statement(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/context.py", line 306, in orm_execute_statement
    result = conn.execute(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1415, in execute
    return meth(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/sql/elements.py", line 523, in _execute_on_connection
    return connection._execute_clauseelement(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1637, in _execute_clauseelement
    ret = self._execute_context(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1842, in _execute_context
    return self._exec_single_context(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1982, in _exec_single_context
    self._handle_dbapi_exception(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 2351, in _handle_dbapi_exception
    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1963, in _exec_single_context
    self.dialect.do_execute(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/default.py", line 943, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedColumn) column "locked_at" of relation "urls" does not exist
LINE 1: UPDATE urls SET status='crawling', locked_at='2025-07-10T07:...
                                           ^

[SQL: UPDATE urls SET status=%(status)s, locked_at=%(locked_at)s WHERE urls.id IN (%(id_1_1)s, %(id_1_2)s)]
[parameters: {'status': 'crawling', 'locked_at': datetime.datetime(2025, 7, 10, 7, 37, 31, 871244, tzinfo=datetime.timezone.utc), 'id_1_1': 1, 'id_1_2': 2}]
(Background on this error at: https://sqlalche.me/e/20/f405)
2025-07-10 13:07:39,242 - worker - ERROR - Reaper: Failed to reset stalled jobs: (psycopg2.errors.UndefinedColumn) column urls.locked_at does not exist
LINE 1: ...RE urls.status IN ('classifying', 'crawling') AND urls.locke...
                                                             ^

[SQL: UPDATE urls SET status=CASE WHEN (urls.status = %(status_1)s) THEN %(param_1)s WHEN (urls.status = %(status_2)s) THEN %(param_2)s END, locked_at=%(locked_at)s WHERE urls.status IN (%(status_3_1)s, %(status_3_2)s) AND urls.locked_at < %(locked_at_1)s]
[parameters: {'status_1': 'classifying', 'param_1': 'pending_classification', 'status_2': 'crawling', 'param_2': 'pending_crawl', 'locked_at': None, 'locked_at_1': datetime.datetime(2025, 7, 10, 7, 22, 38, 412685, tzinfo=datetime.timezone.utc), 'status_3_1': 'classifying', 'status_3_2': 'crawling'}]
(Background on this error at: https://sqlalche.me/e/20/f405)
Traceback (most recent call last):
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1963, in _exec_single_context
    self.dialect.do_execute(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/default.py", line 943, in do_execute
    cursor.execute(statement, parameters)
psycopg2.errors.UndefinedColumn: column urls.locked_at does not exist
LINE 1: ...RE urls.status IN ('classifying', 'crawling') AND urls.locke...
                                                             ^


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/ayush/Desktop/code/blogspy_predictor/src/worker.py", line 289, in reset_stalled_jobs
    rows_reset = stalled_jobs_query.update(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/query.py", line 3299, in update
    result: CursorResult[Any] = self.session.execute(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 2365, in execute
    return self._execute_internal(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 2251, in _execute_internal
    result: Result[Any] = compile_state_cls.orm_execute_statement(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/bulk_persistence.py", line 1648, in orm_execute_statement
    return super().orm_execute_statement(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/context.py", line 306, in orm_execute_statement
    result = conn.execute(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1415, in execute
    return meth(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/sql/elements.py", line 523, in _execute_on_connection
    return connection._execute_clauseelement(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1637, in _execute_clauseelement
    ret = self._execute_context(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1842, in _execute_context
    return self._exec_single_context(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1982, in _exec_single_context
    self._handle_dbapi_exception(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 2351, in _handle_dbapi_exception
    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1963, in _exec_single_context
    self.dialect.do_execute(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/default.py", line 943, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedColumn) column urls.locked_at does not exist
LINE 1: ...RE urls.status IN ('classifying', 'crawling') AND urls.locke...
                                                             ^

[SQL: UPDATE urls SET status=CASE WHEN (urls.status = %(status_1)s) THEN %(param_1)s WHEN (urls.status = %(status_2)s) THEN %(param_2)s END, locked_at=%(locked_at)s WHERE urls.status IN (%(status_3_1)s, %(status_3_2)s) AND urls.locked_at < %(locked_at_1)s]
[parameters: {'status_1': 'classifying', 'param_1': 'pending_classification', 'status_2': 'crawling', 'param_2': 'pending_crawl', 'locked_at': None, 'locked_at_1': datetime.datetime(2025, 7, 10, 7, 22, 38, 412685, tzinfo=datetime.timezone.utc), 'status_3_1': 'classifying', 'status_3_2': 'crawling'}]
(Background on this error at: https://sqlalche.me/e/20/f405)
2025-07-10 14:37:06,937 - worker - INFO - --- Starting BlogSpy Worker ---
2025-07-10 14:37:07,310 - worker - INFO - Model loaded successfully.
2025-07-10 14:37:07,310 - worker - INFO - Initialized with BATCH_SIZE=500 and max_workers=80.
2025-07-10 14:37:15,888 - worker - ERROR - Reaper: Failed to reset stalled jobs: (psycopg2.errors.DatatypeMismatch) column "status" is of type crawl_status but expression is of type text
LINE 1: UPDATE urls SET status=CASE WHEN (urls.status = 'classifying...
                               ^
HINT:  You will need to rewrite or cast the expression.

[SQL: UPDATE urls SET status=CASE WHEN (urls.status = %(status_1)s) THEN %(param_1)s WHEN (urls.status = %(status_2)s) THEN %(param_2)s END, locked_at=%(locked_at)s WHERE urls.status IN (%(status_3_1)s, %(status_3_2)s) AND urls.locked_at < %(locked_at_1)s]
[parameters: {'status_1': 'classifying', 'param_1': 'pending_classification', 'status_2': 'crawling', 'param_2': 'pending_crawl', 'locked_at': None, 'locked_at_1': datetime.datetime(2025, 7, 10, 8, 52, 7, 413934, tzinfo=datetime.timezone.utc), 'status_3_1': 'classifying', 'status_3_2': 'crawling'}]
(Background on this error at: https://sqlalche.me/e/20/f405)
Traceback (most recent call last):
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1963, in _exec_single_context
    self.dialect.do_execute(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/default.py", line 943, in do_execute
    cursor.execute(statement, parameters)
psycopg2.errors.DatatypeMismatch: column "status" is of type crawl_status but expression is of type text
LINE 1: UPDATE urls SET status=CASE WHEN (urls.status = 'classifying...
                               ^
HINT:  You will need to rewrite or cast the expression.


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/ayush/Desktop/code/blogspy_predictor/src/worker.py", line 289, in reset_stalled_jobs
    rows_reset = stalled_jobs_query.update(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/query.py", line 3299, in update
    result: CursorResult[Any] = self.session.execute(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 2365, in execute
    return self._execute_internal(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/session.py", line 2251, in _execute_internal
    result: Result[Any] = compile_state_cls.orm_execute_statement(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/bulk_persistence.py", line 1648, in orm_execute_statement
    return super().orm_execute_statement(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/orm/context.py", line 306, in orm_execute_statement
    result = conn.execute(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1415, in execute
    return meth(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/sql/elements.py", line 523, in _execute_on_connection
    return connection._execute_clauseelement(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1637, in _execute_clauseelement
    ret = self._execute_context(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1842, in _execute_context
    return self._exec_single_context(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1982, in _exec_single_context
    self._handle_dbapi_exception(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 2351, in _handle_dbapi_exception
    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1963, in _exec_single_context
    self.dialect.do_execute(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/engine/default.py", line 943, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.DatatypeMismatch) column "status" is of type crawl_status but expression is of type text
LINE 1: UPDATE urls SET status=CASE WHEN (urls.status = 'classifying...
                               ^
HINT:  You will need to rewrite or cast the expression.

[SQL: UPDATE urls SET status=CASE WHEN (urls.status = %(status_1)s) THEN %(param_1)s WHEN (urls.status = %(status_2)s) THEN %(param_2)s END, locked_at=%(locked_at)s WHERE urls.status IN (%(status_3_1)s, %(status_3_2)s) AND urls.locked_at < %(locked_at_1)s]
[parameters: {'status_1': 'classifying', 'param_1': 'pending_classification', 'status_2': 'crawling', 'param_2': 'pending_crawl', 'locked_at': None, 'locked_at_1': datetime.datetime(2025, 7, 10, 8, 52, 7, 413934, tzinfo=datetime.timezone.utc), 'status_3_1': 'classifying', 'status_3_2': 'crawling'}]
(Background on this error at: https://sqlalche.me/e/20/f405)
2025-07-10 14:37:18,765 - worker - INFO - Locked and dispatched 2 crawling jobs.
2025-07-10 14:37:23,688 - worker - INFO - ✅ (From Classify) Crawl logic for https://goyalayus.github.io/ added 13 new URLs.
2025-07-10 14:39:25,499 - worker - INFO - --- Starting BlogSpy Worker ---
2025-07-10 14:39:25,854 - worker - INFO - Model loaded successfully.
2025-07-10 14:39:25,855 - worker - INFO - Initialized with BATCH_SIZE=500 and max_workers=80.
2025-07-10 14:39:25,957 - worker - ERROR - Reaper: Failed to reset stalled jobs: EnumMeta.__call__() missing 1 required positional argument: 'value'
Traceback (most recent call last):
  File "/home/ayush/Desktop/code/blogspy_predictor/src/worker.py", line 280, in reset_stalled_jobs
    (URL.status == 'classifying', cast(
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/sql/_elements_constructors.py", line 912, in cast
    return Cast(expression, type_)
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/sql/elements.py", line 3430, in __init__
    self.type = type_api.to_instance(type_)
  File "/home/ayush/.local/lib/python3.10/site-packages/sqlalchemy/sql/type_api.py", line 2336, in to_instance
    return typeobj(*arg, **kw)
TypeError: EnumMeta.__call__() missing 1 required positional argument: 'value'
2025-07-10 14:41:37,659 - worker - INFO - --- Starting BlogSpy Worker ---
2025-07-10 14:41:38,013 - worker - INFO - Model loaded successfully.
2025-07-10 14:41:38,013 - worker - INFO - Initialized with BATCH_SIZE=500 and max_workers=80.
2025-07-10 14:41:50,041 - worker - INFO - Locked and dispatched 2 crawling jobs.
2025-07-10 14:41:54,145 - worker - INFO - ✅ (From Classify) Crawl logic for https://goyalayus.github.io/ added 13 new URLs.
2025-07-10 14:41:57,840 - worker - INFO - ✅ (From Classify) Crawl logic for https://paulgraham.com/ added 3 new URLs.
2025-07-10 14:41:58,662 - worker - INFO - Batch of 2 crawling jobs complete.
2025-07-10 14:42:00,096 - worker - INFO - Locked and dispatched 11 classification jobs.
2025-07-10 14:42:03,806 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/goyalayus added 6 new URLs.
2025-07-10 14:42:05,408 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/Sikandar_lx added 6 new URLs.
2025-07-10 14:42:05,413 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/goyalayus/status/1922024740639682799 added 6 new URLs.
2025-07-10 14:42:05,683 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/arnnnvvv added 6 new URLs.
2025-07-10 14:42:06,911 - worker - INFO - Batch of 11 classification jobs complete.
2025-07-10 14:42:09,915 - worker - INFO - Locked and dispatched 5 crawling jobs.
2025-07-10 14:42:12,817 - worker - INFO - ✅ (From Classify) Crawl logic for https://goyalayus.github.io/index.html added 0 new URLs.
2025-07-10 14:42:12,819 - worker - INFO - ✅ (From Classify) Crawl logic for https://goyalayus.github.io/peeple.html added 0 new URLs.
2025-07-10 14:42:13,726 - worker - INFO - ✅ (From Classify) Crawl logic for https://paulgraham.com/goodwriting.html added 1 new URLs.
2025-07-10 14:42:13,726 - worker - INFO - ✅ (From Classify) Crawl logic for https://goyalayus.github.io/blog.html added 9 new URLs.
2025-07-10 14:42:15,073 - worker - INFO - ✅ (From Classify) Crawl logic for https://paulgraham.com/foundermode.html added 1 new URLs.
2025-07-10 14:42:15,889 - worker - INFO - Batch of 5 crawling jobs complete.
2025-07-10 14:42:17,322 - worker - INFO - Locked and dispatched 6 classification jobs.
2025-07-10 14:42:18,973 - worker - ERROR - Failed to classify https://help.x.com/using-x/x-supported-browsers: 403 Client Error: Forbidden for url: https://help.x.com/using-x/x-supported-browsers
2025-07-10 14:42:20,748 - worker - ERROR - Failed to classify https://support.twitter.com/articles/20170514: 403 Client Error: Forbidden for url: https://help.x.com/articles/20170514
2025-07-10 14:42:23,864 - worker - INFO - ✅ (From Classify) Crawl logic for https://legal.twitter.com/imprint.html added 10 new URLs.
2025-07-10 14:42:24,584 - worker - INFO - Batch of 6 classification jobs complete.
2025-07-10 14:42:27,093 - worker - INFO - Locked and dispatched 12 crawling jobs.
2025-07-10 14:42:30,152 - worker - INFO - ✅ (From Classify) Crawl logic for https://goyalayus.github.io/blog/imagegen-skill-issue.html added 0 new URLs.
2025-07-10 14:42:30,292 - worker - INFO - ✅ (From Classify) Crawl logic for https://goyalayus.github.io/blog/how-llm-tokenizers-work-bpe.html added 0 new URLs.
2025-07-10 14:42:30,487 - worker - INFO - ✅ (From Classify) Crawl logic for https://goyalayus.github.io/blog/students-who-stammer.html added 0 new URLs.
2025-07-10 14:42:31,337 - worker - INFO - ✅ (From Classify) Crawl logic for https://paulgraham.com/index.html added 0 new URLs.
2025-07-10 14:42:34,573 - worker - INFO - ✅ (From Classify) Crawl logic for https://goyalayus.github.io/blog/decoding-transformers.html added 1 new URLs.
2025-07-10 14:42:36,415 - worker - INFO - ✅ (From Classify) Crawl logic for https://goyalayus.github.io/blog/summer-goals.html added 2 new URLs.
2025-07-10 14:42:36,489 - worker - INFO - ✅ (From Classify) Crawl logic for https://goyalayus.github.io/blog/startup-context.html added 26 new URLs.
2025-07-10 14:42:40,026 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/tos added 99 new URLs.
2025-07-10 14:42:40,712 - worker - INFO - ✅ (From Classify) Crawl logic for https://goyalayus.github.io/blog/kv-caching-attention.html added 1 new URLs.
2025-07-10 14:42:40,793 - worker - INFO - ✅ (From Classify) Crawl logic for https://goyalayus.github.io/blog/two-startups.html added 1 new URLs.
2025-07-10 14:42:40,903 - worker - INFO - ✅ (From Classify) Crawl logic for https://goyalayus.github.io/blog/distribution-for-pmf.html added 2 new URLs.
2025-07-10 14:42:45,388 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/privacy added 115 new URLs.
2025-07-10 14:42:47,331 - worker - INFO - Batch of 12 crawling jobs complete.
2025-07-10 14:42:48,925 - worker - INFO - Locked and dispatched 114 classification jobs.
2025-07-10 14:42:51,710 - worker - ERROR - Failed to classify https://help.x.com/forms/privacy: 403 Client Error: Forbidden for url: https://help.x.com/forms/privacy
2025-07-10 14:42:51,746 - worker - ERROR - Failed to classify https://help.x.com/safety-and-security/remove-x-profile-from-google-search: 403 Client Error: Forbidden for url: https://help.x.com/safety-and-security/remove-x-profile-from-google-search
2025-07-10 14:42:51,752 - worker - ERROR - Failed to classify https://help.x.com/rules-and-policies/information-for-parents-and-minor-users: 403 Client Error: Forbidden for url: https://help.x.com/rules-and-policies/information-for-parents-and-minor-users
2025-07-10 14:42:52,038 - worker - ERROR - Failed to classify https://help.x.com/managing-your-account/how-to-download-your-x-archive: 403 Client Error: Forbidden for url: https://help.x.com/managing-your-account/how-to-download-your-x-archive
2025-07-10 14:42:52,724 - worker - ERROR - Failed to classify https://help.x.com/using-x/spaces: 403 Client Error: Forbidden for url: https://help.x.com/using-x/spaces
2025-07-10 14:42:52,878 - worker - ERROR - Failed to classify https://help.x.com/rules-and-policies/x-law-enforcement-support: 403 Client Error: Forbidden for url: https://help.x.com/rules-and-policies/x-law-enforcement-support
2025-07-10 14:42:53,014 - worker - ERROR - Failed to classify https://help.x.com/safety-and-security/data-through-partnerships: 403 Client Error: Forbidden for url: https://help.x.com/safety-and-security/data-through-partnerships
2025-07-10 14:42:53,034 - worker - ERROR - Failed to classify https://help.x.com/rules-and-policies/x-rules: 403 Client Error: Forbidden for url: https://help.x.com/rules-and-policies/x-rules
2025-07-10 14:42:53,803 - worker - ERROR - Failed to classify https://help.twitter.com/forms/privacy: 403 Client Error: Forbidden for url: https://help.x.com/forms/privacy
2025-07-10 14:42:56,830 - worker - ERROR - Failed to classify https://help.x.com/rules-and-policies/x-api: 403 Client Error: Forbidden for url: https://help.x.com/rules-and-policies/x-api
2025-07-10 14:42:56,855 - worker - ERROR - Failed to classify https://help.x.com/rules-and-policies/uk-information-for-parents-and-minor-users: 403 Client Error: Forbidden for url: https://help.x.com/rules-and-policies/uk-information-for-parents-and-minor-users
2025-07-10 14:42:57,030 - worker - ERROR - Failed to classify https://help.x.com/forms: 403 Client Error: Forbidden for url: https://help.x.com/forms
2025-07-10 14:42:57,406 - worker - ERROR - Failed to classify https://help.x.com/rules-and-policies/digital-services-act: 403 Client Error: Forbidden for url: https://help.x.com/rules-and-policies/digital-services-act
2025-07-10 14:42:57,511 - worker - ERROR - Failed to classify https://help.x.com/rules-and-policies/twitter-cookies: 403 Client Error: Forbidden for url: https://help.x.com/rules-and-policies/twitter-cookies
2025-07-10 14:42:57,720 - worker - ERROR - Failed to classify https://help.x.com/rules-and-policies/misuse-of-reporting-features: 403 Client Error: Forbidden for url: https://help.x.com/rules-and-policies/misuse-of-reporting-features
2025-07-10 14:42:57,857 - worker - ERROR - Failed to classify https://help.x.com/using-x/direct-messages: 403 Client Error: Forbidden for url: https://help.x.com/using-x/direct-messages
2025-07-10 14:42:58,468 - worker - ERROR - Failed to classify https://help.x.com/rules-and-policies/x-cookies: 403 Client Error: Forbidden for url: https://help.x.com/rules-and-policies/x-cookies
2025-07-10 14:42:58,617 - worker - ERROR - Failed to classify https://help.x.com/rules-and-policies/x-report-violation: 403 Client Error: Forbidden for url: https://help.x.com/rules-and-policies/x-report-violation
2025-07-10 14:42:58,891 - worker - ERROR - Failed to classify https://help.x.com/managing-your-account/suspended-x-accounts: 403 Client Error: Forbidden for url: https://help.x.com/managing-your-account/suspended-x-accounts
2025-07-10 14:42:59,180 - worker - ERROR - Failed to classify https://help.x.com/managing-your-account/connect-or-revoke-access-to-third-party-apps?lang=browser: 403 Client Error: Forbidden for url: https://help.x.com/managing-your-account/connect-or-revoke-access-to-third-party-apps?lang=browser
2025-07-10 14:42:59,213 - worker - ERROR - Failed to classify https://help.x.com/forms/account-access/appeals: 403 Client Error: Forbidden for url: https://help.x.com/forms/account-access/appeals
2025-07-10 14:42:59,723 - worker - ERROR - Failed to classify https://www.dataprivacyframework.gov/framework-article/ANNEX-I-introduction: HTTPSConnectionPool(host='www.dataprivacyframework.gov', port=443): Read timed out. (read timeout=2)
2025-07-10 14:42:59,927 - worker - ERROR - Failed to classify https://help.x.com/using-x/x-videos: 403 Client Error: Forbidden for url: https://help.x.com/using-x/x-videos
2025-07-10 14:42:59,934 - worker - ERROR - Failed to classify https://help.x.com/rules-and-policies/enforcement-options: 403 Client Error: Forbidden for url: https://help.x.com/rules-and-policies/enforcement-options
2025-07-10 14:42:59,968 - worker - ERROR - Failed to classify https://help.x.com/resources/recommender-systems: 403 Client Error: Forbidden for url: https://help.x.com/resources/recommender-systems
2025-07-10 14:42:59,971 - worker - ERROR - Failed to classify https://www.dataprivacyframework.gov/s/: HTTPSConnectionPool(host='www.dataprivacyframework.gov', port=443): Read timed out. (read timeout=2)
2025-07-10 14:43:00,164 - worker - ERROR - Failed to classify https://www.dataprivacyframework.gov/list: HTTPSConnectionPool(host='www.dataprivacyframework.gov', port=443): Read timed out. (read timeout=2)
2025-07-10 14:43:00,327 - worker - ERROR - Failed to classify https://help.x.com/resources/accessibility: 403 Client Error: Forbidden for url: https://help.x.com/resources/accessibility
2025-07-10 14:43:00,419 - worker - ERROR - Failed to classify https://help.x.com/rules-and-policies/data-processing-legal-bases: 403 Client Error: Forbidden for url: https://help.x.com/rules-and-policies/data-processing-legal-bases
2025-07-10 14:43:00,465 - worker - ERROR - Failed to classify https://help.x.com/rules-and-policies/platform-manipulation: 403 Client Error: Forbidden for url: https://help.x.com/rules-and-policies/platform-manipulation
2025-07-10 14:43:00,561 - worker - ERROR - Failed to classify https://marketing.x.com/en/success-stories: HTTPSConnectionPool(host='marketing.x.com', port=443): Read timed out. (read timeout=2)
2025-07-10 14:43:00,638 - worker - ERROR - Failed to classify https://help.x.com/using-x/encrypted-direct-messages: 403 Client Error: Forbidden for url: https://help.x.com/using-x/encrypted-direct-messages
2025-07-10 14:43:00,647 - worker - ERROR - Failed to classify https://help.x.com/: 403 Client Error: Forbidden for url: https://help.x.com/
2025-07-10 14:43:00,708 - worker - ERROR - Failed to classify https://help.x.com/forms/dmca: 403 Client Error: Forbidden for url: https://help.x.com/forms/dmca
2025-07-10 14:43:00,754 - worker - ERROR - Failed to classify https://help.x.com/en/managing-your-account: 403 Client Error: Forbidden for url: https://help.x.com/en/managing-your-account
2025-07-10 14:43:01,129 - worker - ERROR - Failed to classify https://help.x.com/en/rules-and-policies: 403 Client Error: Forbidden for url: https://help.x.com/en/rules-and-policies
2025-07-10 14:43:01,212 - worker - ERROR - Failed to classify https://developer.x.com/en/docs: HTTPSConnectionPool(host='docs.x.com', port=443): Read timed out. (read timeout=2)
2025-07-10 14:43:01,392 - worker - ERROR - Failed to classify https://help.x.com/managing-your-account/how-to-deactivate-x-account: 403 Client Error: Forbidden for url: https://help.x.com/managing-your-account/how-to-deactivate-x-account
2025-07-10 14:43:01,453 - worker - ERROR - Failed to classify https://help.x.com/rules-and-policies: 403 Client Error: Forbidden for url: https://help.x.com/rules-and-policies
2025-07-10 14:43:01,766 - worker - ERROR - Failed to classify https://grok.com: 403 Client Error: Forbidden for url: https://grok.com/
2025-07-10 14:43:01,895 - worker - ERROR - Failed to classify https://help.x.com/de/using-twitter/twitter-timeline.html: 403 Client Error: Forbidden for url: https://help.x.com/de/using-twitter/twitter-timeline.html
2025-07-10 14:43:02,177 - worker - ERROR - Failed to classify https://help.x.com/en/using-twitter: 403 Client Error: Forbidden for url: https://help.x.com/en/using-twitter
2025-07-10 14:43:02,205 - worker - ERROR - Failed to classify https://investor.twitterinc.com/: 403 Client Error: Forbidden for url: https://investor.twitterinc.com/
2025-07-10 14:43:02,321 - worker - INFO - Batch progress: 20/114 classification jobs finished.
2025-07-10 14:43:02,428 - worker - ERROR - Failed to classify https://developer.x.com/docs/x-for-websites/cards/overview/abouts-card: 404 Client Error: Not Found for url: https://developer.x.com/en/docs/x-for-websites/cards/overview/abouts-card
2025-07-10 14:43:02,652 - worker - ERROR - Failed to classify https://help.x.com/en/forms.html: 403 Client Error: Forbidden for url: https://help.x.com/en/forms.html
2025-07-10 14:43:02,715 - worker - ERROR - Failed to classify https://help.x.com/rules-and-policies/x-services-and-corporate-affiliates: HTTPSConnectionPool(host='help.x.com', port=443): Read timed out. (read timeout=4)
2025-07-10 14:43:02,881 - worker - ERROR - Failed to classify https://help.x.com/using-x/download-the-x-app: 403 Client Error: Forbidden for url: https://help.x.com/using-x/download-the-x-app
2025-07-10 14:43:03,143 - worker - ERROR - Failed to classify https://help.x.com/using-x/parental-consent: HTTPSConnectionPool(host='help.x.com', port=443): Read timed out. (read timeout=4)
2025-07-10 14:43:03,506 - worker - ERROR - Failed to classify https://commission.europa.eu/law/law-topic/data-protection/international-dimension-data-protection/standard-contractual-clauses-scc_en: HTTPSConnectionPool(host='commission.europa.eu', port=443): Read timed out. (read timeout=4)
2025-07-10 14:43:04,557 - worker - INFO - Batch progress: 40/114 classification jobs finished.
2025-07-10 14:43:05,465 - worker - INFO - Batch progress: 60/114 classification jobs finished.
2025-07-10 14:43:05,581 - worker - INFO - ✅ (From Classify) Crawl logic for https://legal.x.com/cdn-cgi/l/email-protection added 4 new URLs.
2025-07-10 14:43:05,606 - worker - INFO - ✅ (From Classify) Crawl logic for https://legal.x.com/de/imprint.html added 0 new URLs.
2025-07-10 14:43:05,608 - worker - ERROR - Failed to classify https://help.twitter.com/rules-and-policies/twitter-cookies: 403 Client Error: Forbidden for url: https://help.x.com/rules-and-policies/twitter-cookies
2025-07-10 14:43:07,294 - worker - ERROR - Failed to classify https://blog.x.com/engineering/en_us.html: 403 Client Error: Forbidden for url: https://blog.x.com/engineering/en_us.html
2025-07-10 14:43:07,318 - worker - ERROR - Failed to classify https://blog.x.com/: 403 Client Error: Forbidden for url: https://blog.x.com/
2025-07-10 14:43:07,431 - worker - ERROR - Failed to classify https://blog.x.com/developer/: 403 Client Error: Forbidden for url: https://blog.x.com/developer/
2025-07-10 14:43:07,556 - worker - INFO - Batch progress: 80/114 classification jobs finished.
2025-07-10 14:43:07,712 - worker - ERROR - Failed to classify https://support.twitter.com/forms: 403 Client Error: Forbidden for url: https://help.x.com/forms
2025-07-10 14:43:07,983 - worker - INFO - ✅ (From Classify) Crawl logic for https://legal.x.com/tr/imprint.html added 2 new URLs.
2025-07-10 14:43:08,495 - worker - INFO - ✅ (From Classify) Crawl logic for https://twitter.com added 0 new URLs.
2025-07-10 14:43:09,416 - worker - ERROR - Failed to classify https://about.x.com/en/who-we-are/our-company.html: 404 Client Error: Not Found for url: https://about.x.com/en/who-we-are/our-company
2025-07-10 14:43:10,060 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tam-notes-thoughts/ added 188 new URLs.
2025-07-10 14:43:10,836 - worker - INFO - Batch progress: 100/114 classification jobs finished.
2025-07-10 14:43:11,137 - worker - ERROR - Failed to classify https://developer.x.com/docs: (psycopg2.errors.DeadlockDetected) deadlock detected
DETAIL:  Process 1629 waits for ShareLock on transaction 22692; blocked by process 1603.
Process 1603 waits for ShareLock on transaction 22691; blocked by process 1629.
HINT:  See server log for query details.
CONTEXT:  while inserting index tuple (8,21) in relation "urls"

[SQL: INSERT INTO urls (url, netloc, status, rendering) VALUES (%(url_m0)s, %(netloc_m0)s, %(status_m0)s, %(rendering)s), (%(url_m1)s, %(netloc_m1)s, %(status_m1)s, %(rendering_m1)s) ON CONFLICT (url) DO NOTHING]
[parameters: {'url_m0': 'https://developer.x.com/en/support', 'netloc_m0': 'developer.x.com', 'status_m0': 'pending_crawl', 'rendering': 'UNKNOWN', 'url_m1': 'https://developer.x.com/en/portal/petition/essential/basic-info', 'netloc_m1': 'developer.x.com', 'status_m1': 'pending_crawl', 'rendering_m1': 'UNKNOWN'}]
(Background on this error at: https://sqlalche.me/e/20/e3q8)
2025-07-10 14:43:12,056 - worker - ERROR - Transaction failed for job ID 121. Rolling back. Error: (psycopg2.errors.InFailedSqlTransaction) current transaction is aborted, commands ignored until end of transaction block

[SQL: UPDATE urls SET status=%(status)s, rendering=%(rendering)s, content=%(content)s, title=%(title)s, error_message=%(error_message)s, processed_at=%(processed_at)s WHERE urls.id = %(urls_id)s]
[parameters: {'status': 'failed', 'rendering': 'SSR', 'content': "Home - X X home page English Search... ⌘ K Ask AI Support Developer portal Developer portal Search... Navigation Home Get started X API X Ads API Tut ... (928 characters truncated) ...  it and get inspired to use the APIs in your own way. Browse all success stories Assistant Responses are generated using AI and may contain mistakes.", 'title': 'Home - X', 'error_message': '(psycopg2.errors.DeadlockDetected) deadlock detected\nDETAIL:  Process 1629 waits for ShareLock on transaction 22692; blocked by process 1603.\nProce ... (658 characters truncated) ...  \'developer.x.com\', \'status_m1\': \'pending_crawl\', \'rendering_m1\': \'UNKNOWN\'}]\n(Background on this error at: https://sqlalche.me/e/20/e3q8)', 'processed_at': datetime.datetime(2025, 7, 10, 9, 13, 7, 573299, tzinfo=datetime.timezone.utc), 'urls_id': 121}]
(Background on this error at: https://sqlalche.me/e/20/2j85)
2025-07-10 14:43:12,910 - worker - INFO - ✅ (From Classify) Crawl logic for https://docs.x.com/resources/fundamentals/authentication/guides/log-in-with-x added 37 new URLs.
2025-07-10 14:43:13,826 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/the-busy-founders-guide-to-pmf/ added 183 new URLs.
2025-07-10 14:43:14,076 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/the-indus-valley-playbook/ added 192 new URLs.
2025-07-10 14:43:15,259 - worker - INFO - ✅ (From Classify) Crawl logic for https://docs.x.com/status added 16 new URLs.
2025-07-10 14:43:16,440 - worker - INFO - Batch of 114 classification jobs complete.
2025-07-10 14:43:19,598 - worker - INFO - Locked and dispatched 312 crawling jobs.
2025-07-10 14:43:25,104 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/settings/autoplay added 0 new URLs.
2025-07-10 14:43:26,304 - worker - ERROR - Failed to crawl https://x.com/hu/privacy: HTTPSConnectionPool(host='x.com', port=443): Read timed out. (read timeout=2)
2025-07-10 14:43:26,339 - worker - ERROR - Failed to crawl http://paulgraham.com/makersschedule.html: HTTPSConnectionPool(host='paulgraham.com', port=443): Read timed out. (read timeout=2)
2025-07-10 14:43:26,341 - worker - ERROR - Failed to crawl http://paulgraham.com/ideas.html: HTTPSConnectionPool(host='paulgraham.com', port=443): Read timed out. (read timeout=2)
2025-07-10 14:43:26,380 - worker - ERROR - Failed to crawl http://paulgraham.com/really.html: HTTPSConnectionPool(host='paulgraham.com', port=443): Read timed out. (read timeout=2)
2025-07-10 14:43:28,319 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/content/twitter-com/legal/km added 0 new URLs.
2025-07-10 14:43:28,488 - worker - ERROR - Failed to crawl https://x.com/content/twitter-com/legal/fa: HTTPSConnectionPool(host='x.com', port=443): Read timed out. (read timeout=4)
2025-07-10 14:43:28,553 - worker - ERROR - Failed to crawl https://x.com/ga/tos: HTTPSConnectionPool(host='x.com', port=443): Read timed out. (read timeout=4)
2025-07-10 14:43:28,566 - worker - ERROR - Failed to crawl https://x.com/ja/privacy: HTTPSConnectionPool(host='x.com', port=443): Read timed out. (read timeout=4)
2025-07-10 14:43:28,600 - worker - ERROR - Failed to crawl https://x.com/content/twitter-com/legal/zh-cn: HTTPSConnectionPool(host='x.com', port=443): Read timed out. (read timeout=4)
2025-07-10 14:43:28,601 - worker - ERROR - Failed to crawl https://x.com/pl/privacy: HTTPSConnectionPool(host='x.com', port=443): Read timed out. (read timeout=4)
2025-07-10 14:43:28,633 - worker - ERROR - Failed to crawl https://x.com/privacy/previous: HTTPSConnectionPool(host='x.com', port=443): Read timed out. (read timeout=4)
2025-07-10 14:43:28,913 - worker - ERROR - Failed to crawl https://x.com/hmn/tos: HTTPSConnectionPool(host='x.com', port=443): Read timed out. (read timeout=4)
2025-07-10 14:43:28,915 - worker - ERROR - Failed to crawl https://x.com/cs/tos: HTTPSConnectionPool(host='x.com', port=443): Read timed out. (read timeout=4)
2025-07-10 14:43:29,162 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/waitin4agi_ added 0 new URLs.
2025-07-10 14:43:29,355 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/content/twitter-com/legal/hy added 0 new URLs.
2025-07-10 14:43:29,601 - worker - ERROR - Failed to crawl https://x.com/pl/tos: HTTPSConnectionPool(host='x.com', port=443): Read timed out. (read timeout=4)
2025-07-10 14:43:29,893 - worker - ERROR - Failed to crawl https://x.com/zh-cn/tos: HTTPSConnectionPool(host='x.com', port=443): Read timed out. (read timeout=4)
2025-07-10 14:43:29,898 - worker - ERROR - Failed to crawl http://paulgraham.com/start.html: HTTPSConnectionPool(host='paulgraham.com', port=443): Read timed out. (read timeout=4)
2025-07-10 14:43:29,898 - worker - ERROR - Failed to crawl http://paulgraham.com/startupmistakes.html: HTTPSConnectionPool(host='paulgraham.com', port=443): Read timed out. (read timeout=4)
2025-07-10 14:43:29,915 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/content/twitter-com/legal/tl added 0 new URLs.
2025-07-10 14:43:30,142 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/cdn-cgi/l/email-protection added 0 new URLs.
2025-07-10 14:43:30,175 - worker - ERROR - Failed to crawl https://x.com/lv/tos: HTTPSConnectionPool(host='x.com', port=443): Read timed out. (read timeout=4)
2025-07-10 14:43:30,191 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/content/twitter-com/legal/vi added 0 new URLs.
2025-07-10 14:43:30,317 - worker - INFO - ✅ (From Classify) Crawl logic for https://goyalayus.github.io added 0 new URLs.
2025-07-10 14:43:30,732 - worker - INFO - ✅ (From Classify) Crawl logic for http://paulgraham.com/hackinvest.html added 0 new URLs.
2025-07-10 14:43:31,243 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/settings added 0 new URLs.
2025-07-10 14:43:31,906 - worker - INFO - Batch progress: 20/312 crawling jobs finished.
2025-07-10 14:43:31,938 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/settings/your_twitter_data added 0 new URLs.
2025-07-10 14:43:31,955 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/settings/privacy_and_safety added 0 new URLs.
2025-07-10 14:43:31,985 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/settings/account added 0 new URLs.
2025-07-10 14:43:32,121 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/lv/privacy added 2 new URLs.
2025-07-10 14:43:32,389 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/content/twitter-com/legal/ar added 0 new URLs.
2025-07-10 14:43:32,929 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/content/twitter-com/legal/zh-tw added 0 new URLs.
2025-07-10 14:43:33,518 - worker - INFO - ✅ (From Classify) Crawl logic for http://paulgraham.com/growth.html added 5 new URLs.
2025-07-10 14:43:33,646 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/content/twitter-com/legal/hmn added 0 new URLs.
2025-07-10 14:43:34,612 - worker - INFO - ✅ (From Classify) Crawl logic for http://paulgraham.com/ds.html added 10 new URLs.
2025-07-10 14:43:34,614 - worker - INFO - ✅ (From Classify) Crawl logic for http://paulgraham.com/startupideas.html added 5 new URLs.
2025-07-10 14:43:34,614 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/en/privacy added 0 new URLs.
2025-07-10 14:43:34,616 - worker - INFO - ✅ (From Classify) Crawl logic for http://paulgraham.com/good.html added 6 new URLs.
2025-07-10 14:43:35,719 - worker - ERROR - Failed to crawl https://sajithpai.com/category/venture-capital-startups/: HTTPSConnectionPool(host='sajithpai.com', port=443): Read timed out. (read timeout=2)
2025-07-10 14:43:35,901 - worker - ERROR - Failed to crawl https://sajithpai.com/tag/valuation/: HTTPSConnectionPool(host='sajithpai.com', port=443): Read timed out. (read timeout=2)
2025-07-10 14:43:35,932 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/ added 0 new URLs.
2025-07-10 14:43:36,570 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/ru/privacy added 3 new URLs.
2025-07-10 14:43:36,584 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/hr/privacy added 2 new URLs.
2025-07-10 14:43:36,584 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/zh-tw/tos added 4 new URLs.
2025-07-10 14:43:36,586 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/hu/tos added 2 new URLs.
2025-07-10 14:43:36,588 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/id/privacy added 3 new URLs.
2025-07-10 14:43:36,601 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/sk/privacy added 2 new URLs.
2025-07-10 14:43:36,601 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/fr/tos added 2 new URLs.
2025-07-10 14:43:36,607 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/el/privacy added 2 new URLs.
2025-07-10 14:43:36,609 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/et/tos added 2 new URLs.
2025-07-10 14:43:36,627 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/sv/tos added 2 new URLs.
2025-07-10 14:43:36,640 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/ro/privacy added 2 new URLs.
2025-07-10 14:43:36,676 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/nl/privacy added 2 new URLs.
2025-07-10 14:43:36,681 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/ko/tos added 3 new URLs.
2025-07-10 14:43:36,693 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/lt/tos added 2 new URLs.
2025-07-10 14:43:36,696 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/pt/privacy added 3 new URLs.
2025-07-10 14:43:36,709 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/de/privacy added 2 new URLs.
2025-07-10 14:43:36,712 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/fi/tos added 2 new URLs.
2025-07-10 14:43:36,720 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/sl/tos added 2 new URLs.
2025-07-10 14:43:36,726 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/es/tos added 2 new URLs.
2025-07-10 14:43:38,532 - worker - INFO - Batch progress: 40/312 crawling jobs finished.
2025-07-10 14:43:40,701 - worker - ERROR - Failed to crawl https://sajithpai.com/2021/04/: HTTPSConnectionPool(host='sajithpai.com', port=443): Read timed out. (read timeout=2)
2025-07-10 14:43:41,713 - worker - ERROR - Failed to crawl https://sajithpai.com/2018/01/: HTTPSConnectionPool(host='sajithpai.com', port=443): Read timed out. (read timeout=2)
2025-07-10 14:43:43,739 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/hy/tos added 0 new URLs.
2025-07-10 14:43:46,181 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/tos/previous added 19 new URLs.
2025-07-10 14:43:47,743 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/vi/tos added 0 new URLs.
2025-07-10 14:43:47,864 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/ja/tos added 2 new URLs.
2025-07-10 14:43:47,891 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/fa/tos added 0 new URLs.
2025-07-10 14:43:47,966 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/tl/tos added 0 new URLs.
2025-07-10 14:43:48,040 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/en/tos added 0 new URLs.
2025-07-10 14:43:49,133 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/ar/tos added 2 new URLs.
2025-07-10 14:43:49,625 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/da/privacy added 1 new URLs.
2025-07-10 14:43:49,688 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2024/06/ added 0 new URLs.
2025-07-10 14:43:51,407 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/km/tos added 0 new URLs.
2025-07-10 14:43:51,786 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/startups/ added 18 new URLs.
2025-07-10 14:43:51,825 - worker - ERROR - Failed to crawl https://sajithpai.com/2017/07/: HTTPSConnectionPool(host='sajithpai.com', port=443): Read timed out. (read timeout=2)
2025-07-10 14:43:52,355 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/es/privacy added 2 new URLs.
2025-07-10 14:43:54,340 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/bandra-in-a-box/ added 4 new URLs.
2025-07-10 14:43:54,353 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/bg/privacy added 1 new URLs.
2025-07-10 14:43:54,776 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2013/06/ added 3 new URLs.
2025-07-10 14:43:54,777 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2023/08/ added 2 new URLs.
2025-07-10 14:43:55,586 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/india2-english-tax-building-billion-users/ added 27 new URLs.
2025-07-10 14:43:55,779 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/it/tos added 1 new URLs.
2025-07-10 14:43:56,299 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2023/07/ added 7 new URLs.
2025-07-10 14:43:56,307 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2021/12/ added 10 new URLs.
2025-07-10 14:43:56,357 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/cs/privacy added 1 new URLs.
2025-07-10 14:43:56,909 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/ga/privacy added 1 new URLs.
2025-07-10 14:43:56,934 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/sl/privacy added 2 new URLs.
2025-07-10 14:43:57,038 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/nl/tos added 1 new URLs.
2025-07-10 14:43:57,941 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/fr/privacy added 1 new URLs.
2025-07-10 14:44:01,214 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/el/tos added 1 new URLs.
2025-07-10 14:44:01,431 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2024/02/ added 6 new URLs.
2025-07-10 14:44:01,530 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/i-joined-amit-varma-on-the-seen-the-unseen-podcast/ added 4 new URLs.
2025-07-10 14:44:01,889 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2016/01/ added 2 new URLs.
2025-07-10 14:44:01,998 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/mt/privacy added 1 new URLs.
2025-07-10 14:44:02,445 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/sv/privacy added 1 new URLs.
2025-07-10 14:44:03,063 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/category/pmf-convos/ added 19 new URLs.
2025-07-10 14:44:04,311 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/fi/privacy added 2 new URLs.
2025-07-10 14:44:04,318 - worker - INFO - Batch progress: 60/312 crawling jobs finished.
2025-07-10 14:44:04,831 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2019/05/ added 1 new URLs.
2025-07-10 14:44:06,055 - worker - ERROR - Failed to crawl https://sajithpai.com/2021/02/: HTTPSConnectionPool(host='sajithpai.com', port=443): Read timed out. (read timeout=2)
2025-07-10 14:44:08,748 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/et/privacy added 2 new URLs.
2025-07-10 14:44:09,919 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2022/12/ added 1 new URLs.
2025-07-10 14:44:11,067 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/interleavings-notes-thoughts-on-recent-reads-25-may-25/ added 12 new URLs.
2025-07-10 14:44:11,091 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2012/10/ added 20 new URLs.
2025-07-10 14:44:11,224 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2018/05/ added 3 new URLs.
2025-07-10 14:44:11,269 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/pt/tos added 1 new URLs.
2025-07-10 14:44:11,512 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/category/pmf-essays/ added 21 new URLs.
2025-07-10 14:44:11,769 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/ro/tos added 1 new URLs.
2025-07-10 14:44:14,029 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/da/tos added 1 new URLs.
2025-07-10 14:44:14,472 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/de/tos added 1 new URLs.
2025-07-10 14:44:14,831 - worker - INFO - Batch progress: 80/312 crawling jobs finished.
2025-07-10 14:44:15,081 - worker - ERROR - Failed to crawl https://sajithpai.com/2019/02/: HTTPSConnectionPool(host='sajithpai.com', port=443): Read timed out. (read timeout=4)
2025-07-10 14:44:16,039 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/ru/tos added 2 new URLs.
2025-07-10 14:44:16,041 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/it/privacy added 1 new URLs.
2025-07-10 14:44:16,333 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/id/tos added 2 new URLs.
2025-07-10 14:44:16,592 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/hr/tos added 1 new URLs.
2025-07-10 14:44:17,490 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/lt/privacy added 1 new URLs.
2025-07-10 14:44:18,952 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/the-best-vcs-are-long-preloaded-context-windows-for-their-founders/ added 2 new URLs.
2025-07-10 14:44:19,389 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tam-notes-thoughts/?replytocom=3747 added 0 new URLs.
2025-07-10 14:44:20,752 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2025/03/ added 7 new URLs.
2025-07-10 14:44:20,753 - worker - ERROR - Failed to crawl https://sajithpai.com/tag/admissions/: HTTPSConnectionPool(host='sajithpai.com', port=443): Read timed out. (read timeout=4)
2025-07-10 14:44:22,099 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/sk/tos added 1 new URLs.
2025-07-10 14:44:22,196 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2020/08/ added 16 new URLs.
2025-07-10 14:44:22,196 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2020/11/ added 4 new URLs.
2025-07-10 14:44:22,234 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/clayton-christensen/ added 32 new URLs.
2025-07-10 14:44:22,245 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2015/11/ added 8 new URLs.
2025-07-10 14:44:22,928 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/india/ added 8 new URLs.
2025-07-10 14:44:23,248 - worker - ERROR - Failed to crawl https://sajithpai.com/2019/11/: HTTPSConnectionPool(host='sajithpai.com', port=443): Read timed out. (read timeout=2)
2025-07-10 14:44:25,636 - worker - ERROR - Failed to crawl https://sajithpai.com/tag/curation/: HTTPSConnectionPool(host='sajithpai.com', port=443): Max retries exceeded with url: /tag/curation/ (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x7435a45b4460>, 'Connection to sajithpai.com timed out. (connect timeout=2)'))
2025-07-10 14:44:25,954 - worker - INFO - Batch progress: 100/312 crawling jobs finished.
2025-07-10 14:44:26,145 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/metered-model/ added 23 new URLs.
2025-07-10 14:44:26,747 - worker - ERROR - Failed to crawl https://sajithpai.com/2020/09/: HTTPSConnectionPool(host='sajithpai.com', port=443): Max retries exceeded with url: /2020/09/ (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x7435b4721030>, 'Connection to sajithpai.com timed out. (connect timeout=2)'))
2025-07-10 14:44:26,947 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2018/06/ added 0 new URLs.
2025-07-10 14:44:26,972 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/e-commerce/ added 2 new URLs.
2025-07-10 14:44:27,429 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/twitter/ added 12 new URLs.
2025-07-10 14:44:27,931 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2025/05/ added 0 new URLs.
2025-07-10 14:44:28,291 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2018/10/ added 1 new URLs.
2025-07-10 14:44:28,312 - worker - ERROR - Failed to crawl https://sajithpai.com/tag/covid-19/: HTTPSConnectionPool(host='sajithpai.com', port=443): Read timed out. (read timeout=4)
2025-07-10 14:44:28,567 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2020/05/ added 6 new URLs.
2025-07-10 14:44:28,622 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2023/11/ added 1 new URLs.
2025-07-10 14:44:28,673 - worker - ERROR - Failed to crawl https://sajithpai.com/2022/07/: HTTPSConnectionPool(host='sajithpai.com', port=443): Read timed out. (read timeout=4)
2025-07-10 14:44:29,060 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2019/07/ added 3 new URLs.
2025-07-10 14:44:30,056 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2014/12/ added 13 new URLs.
2025-07-10 14:44:30,222 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/category/book-reviews/ added 15 new URLs.
2025-07-10 14:44:30,234 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2023/06/ added 4 new URLs.
2025-07-10 14:44:31,027 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/advertising-revenues/ added 20 new URLs.
2025-07-10 14:44:31,033 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/ko/privacy added 3 new URLs.
2025-07-10 14:44:32,026 - worker - ERROR - Failed to crawl https://sajithpai.com/tag/tam/: HTTPSConnectionPool(host='sajithpai.com', port=443): Read timed out. (read timeout=2)
2025-07-10 14:44:33,013 - worker - ERROR - Failed to crawl https://sajithpai.com/tag/indo-anglians/: HTTPSConnectionPool(host='sajithpai.com', port=443): Max retries exceeded with url: /tag/indo-anglians/ (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x74349a36ee30>, 'Connection to sajithpai.com timed out. (connect timeout=2)'))
2025-07-10 14:44:33,493 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/ugc/ added 56 new URLs.
2025-07-10 14:44:33,627 - worker - INFO - Batch progress: 120/312 crawling jobs finished.
2025-07-10 14:44:34,533 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2018/04/ added 4 new URLs.
2025-07-10 14:44:34,795 - worker - ERROR - Failed to crawl https://sajithpai.com/2016/03/: HTTPSConnectionPool(host='sajithpai.com', port=443): Read timed out. (read timeout=4)
2025-07-10 14:44:35,466 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/online-retail/ added 12 new URLs.
2025-07-10 14:44:35,635 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/social-media/ added 18 new URLs.
2025-07-10 14:44:35,738 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2023/03/ added 7 new URLs.
2025-07-10 14:44:35,880 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2013/10/ added 14 new URLs.
2025-07-10 14:44:36,914 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/content/ added 26 new URLs.
2025-07-10 14:44:37,568 - worker - ERROR - Failed to crawl https://sajithpai.com/2015/03/: HTTPSConnectionPool(host='sajithpai.com', port=443): Read timed out. (read timeout=2)
2025-07-10 14:44:38,146 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2023/10/ added 9 new URLs.
2025-07-10 14:44:38,152 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/usa-newspapers/ added 26 new URLs.
2025-07-10 14:44:38,170 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2024/03/ added 6 new URLs.
2025-07-10 14:44:38,310 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2021/07/ added 1 new URLs.
2025-07-10 14:44:39,116 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2020/12/ added 0 new URLs.
2025-07-10 14:44:40,158 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/mt/tos added 1 new URLs.
2025-07-10 14:44:40,313 - worker - ERROR - Failed to crawl https://sajithpai.com/2014/09/: HTTPSConnectionPool(host='sajithpai.com', port=443): Read timed out. (read timeout=2)
2025-07-10 14:44:40,724 - worker - ERROR - Failed to crawl https://sajithpai.com/2013/08/: HTTPSConnectionPool(host='sajithpai.com', port=443): Read timed out. (read timeout=2)
2025-07-10 14:44:41,244 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2016/09/ added 8 new URLs.
2025-07-10 14:44:41,754 - worker - INFO - Batch progress: 140/312 crawling jobs finished.
2025-07-10 14:44:43,671 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2014/08/ added 12 new URLs.
2025-07-10 14:44:43,699 - worker - ERROR - Failed to crawl https://sajithpai.com/2025/06/: HTTPSConnectionPool(host='sajithpai.com', port=443): Read timed out. (read timeout=2)
2025-07-10 14:44:43,758 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2013/02/ added 11 new URLs.
2025-07-10 14:44:44,204 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2024/09/ added 0 new URLs.
2025-07-10 14:44:47,418 - worker - ERROR - Failed to crawl https://sajithpai.com/2012/11/: HTTPSConnectionPool(host='sajithpai.com', port=443): Read timed out. (read timeout=4)
2025-07-10 14:44:47,442 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2024/01/ added 3 new URLs.
2025-07-10 14:44:47,988 - worker - ERROR - Failed to crawl https://sajithpai.com/2015/01/: HTTPSConnectionPool(host='sajithpai.com', port=443): Read timed out. (read timeout=4)
2025-07-10 14:44:48,252 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2015/08/ added 10 new URLs.
2025-07-10 14:44:48,463 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/higher-education/ added 33 new URLs.
2025-07-10 14:44:48,534 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2021/01/ added 0 new URLs.
2025-07-10 14:44:48,727 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/venture-investing/ added 17 new URLs.
2025-07-10 14:44:49,266 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2021/11/ added 0 new URLs.
2025-07-10 14:44:50,112 - worker - ERROR - Failed to crawl https://sajithpai.com/category/podcast-notes/: HTTPSConnectionPool(host='sajithpai.com', port=443): Read timed out. (read timeout=4)
2025-07-10 14:44:50,199 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/amazon/ added 12 new URLs.
2025-07-10 14:44:50,382 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2024/05/ added 1 new URLs.
2025-07-10 14:44:50,395 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tam-notes-thoughts/?replytocom=3755 added 0 new URLs.
2025-07-10 14:44:50,412 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2020/03/ added 5 new URLs.
2025-07-10 14:44:50,472 - worker - ERROR - Failed to crawl https://sajithpai.com/tag/the-guardian/: HTTPSConnectionPool(host='sajithpai.com', port=443): Read timed out. (read timeout=2)
2025-07-10 14:44:51,327 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/db-corp/ added 33 new URLs.
2025-07-10 14:44:51,755 - worker - INFO - Batch progress: 160/312 crawling jobs finished.
2025-07-10 14:44:52,029 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/book-review/ added 8 new URLs.
2025-07-10 14:44:52,259 - worker - ERROR - Failed to crawl https://sajithpai.com/2023/01/: HTTPSConnectionPool(host='sajithpai.com', port=443): Read timed out. (read timeout=4)
2025-07-10 14:44:53,367 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2012/12/ added 19 new URLs.
2025-07-10 14:44:53,698 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2017/11/ added 0 new URLs.
2025-07-10 14:44:54,164 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2013/04/ added 6 new URLs.
2025-07-10 14:44:54,267 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/classifieds/ added 0 new URLs.
2025-07-10 14:44:54,686 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/preprints/ added 19 new URLs.
2025-07-10 14:44:54,811 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/about-me/ added 6 new URLs.
2025-07-10 14:44:55,283 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/paywalls/ added 3 new URLs.
2025-07-10 14:44:55,284 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2016/05/ added 0 new URLs.
2025-07-10 14:44:56,041 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2013/12/ added 11 new URLs.
2025-07-10 14:44:56,191 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2024/08/ added 1 new URLs.
2025-07-10 14:44:56,426 - worker - INFO - Batch progress: 180/312 crawling jobs finished.
2025-07-10 14:44:56,859 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2018/11/ added 6 new URLs.
2025-07-10 14:44:57,068 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2024/11/ added 13 new URLs.
2025-07-10 14:44:57,509 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/bg/tos added 1 new URLs.
2025-07-10 14:44:58,829 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2023/12/ added 7 new URLs.
2025-07-10 14:44:59,465 - worker - ERROR - Failed to crawl https://sajithpai.com/2021/09/: HTTPSConnectionPool(host='sajithpai.com', port=443): Read timed out. (read timeout=2)
2025-07-10 14:44:59,794 - worker - ERROR - Failed to crawl https://sajithpai.com/tag/news/: HTTPSConnectionPool(host='sajithpai.com', port=443): Read timed out. (read timeout=2)
2025-07-10 14:45:00,145 - worker - ERROR - Failed to crawl https://sajithpai.com/2016/08/: HTTPSConnectionPool(host='sajithpai.com', port=443): Read timed out. (read timeout=2)
2025-07-10 14:45:00,223 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2015/04/ added 17 new URLs.
2025-07-10 14:45:00,530 - worker - ERROR - Failed to crawl https://sajithpai.com/2015/06/: HTTPSConnectionPool(host='sajithpai.com', port=443): Read timed out. (read timeout=2)
2025-07-10 14:45:00,530 - worker - ERROR - Failed to crawl https://sajithpai.com/2022/11/: HTTPSConnectionPool(host='sajithpai.com', port=443): Read timed out. (read timeout=2)
2025-07-10 14:45:01,338 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/circulation-revenue/ added 0 new URLs.
2025-07-10 14:45:01,701 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/advertising-revenue/ added 4 new URLs.
2025-07-10 14:45:01,701 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2012/09/ added 8 new URLs.
2025-07-10 14:45:01,916 - worker - ERROR - Failed to crawl https://sajithpai.com/2024/12/: HTTPSConnectionPool(host='sajithpai.com', port=443): Read timed out. (read timeout=4)
2025-07-10 14:45:02,366 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/the-times-of-india/ added 48 new URLs.
2025-07-10 14:45:02,412 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/washington-post/ added 12 new URLs.
2025-07-10 14:45:02,499 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/advertising/ added 5 new URLs.
2025-07-10 14:45:02,591 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/product-market-fit/ added 3 new URLs.
2025-07-10 14:45:02,963 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/begin-here/ added 10 new URLs.
2025-07-10 14:45:03,063 - worker - ERROR - Failed to crawl https://sajithpai.com/author/sajithpai/: HTTPSConnectionPool(host='sajithpai.com', port=443): Read timed out. (read timeout=2)
2025-07-10 14:45:03,105 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tam-notes-thoughts/?replytocom=3742 added 0 new URLs.
2025-07-10 14:45:03,170 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2018/08/ added 1 new URLs.
2025-07-10 14:45:03,548 - worker - INFO - Batch progress: 200/312 crawling jobs finished.
2025-07-10 14:45:04,549 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2017/10/ added 3 new URLs.
2025-07-10 14:45:04,764 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2012/04/ added 6 new URLs.
2025-07-10 14:45:05,604 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tam-notes-thoughts/?replytocom=3756 added 0 new URLs.
2025-07-10 14:45:05,639 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2023/05/ added 3 new URLs.
2025-07-10 14:45:05,673 - worker - INFO - ✅ (From Classify) Crawl logic for https://x.com/StartupArchive_/status/1779477221788561734 added 0 new URLs.
2025-07-10 14:45:05,839 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2023/04/ added 6 new URLs.
2025-07-10 14:45:06,171 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2017/04/ added 4 new URLs.
2025-07-10 14:45:06,433 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/ashoka-university/ added 6 new URLs.
2025-07-10 14:45:06,443 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2015/05/ added 9 new URLs.
2025-07-10 14:45:06,451 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/strategy/ added 40 new URLs.
2025-07-10 14:45:06,461 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/universities/ added 9 new URLs.
2025-07-10 14:45:07,113 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2014/11/ added 0 new URLs.
2025-07-10 14:45:07,176 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2014/06/ added 21 new URLs.
2025-07-10 14:45:07,461 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2025/01/ added 4 new URLs.
2025-07-10 14:45:07,579 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2020/04/ added 0 new URLs.
2025-07-10 14:45:07,926 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2025/04/ added 1 new URLs.
2025-07-10 14:45:08,781 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/tag/fitness/ added 0 new URLs.
2025-07-10 14:45:08,843 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/2015/10/ added 9 new URLs.
2025-07-10 14:45:09,168 - worker - INFO - ✅ (From Classify) Crawl logic for https://sajithpai.com/how-senior-corporate-executives-exploring-roles-in-venture-or-startups-should-approach-them/ added 3 new URLs.
